File: code-samples/messaging/global-business/cancel-message.js
const MessenteApi = require('messente_api');

const defaultClient = MessenteApi.ApiClient.instance;
const basicAuth = defaultClient.authentications['basicAuth'];
basicAuth.username = 'YOUR_MESSENTE_API_USERNAME';
basicAuth.password = 'YOUR_MESSENTE_API_PASSWORD';

const api = new MessenteApi.OmnimessageApi();

api.cancelScheduledMessage('<omnimessage_id>', (error) => {
  if (error) {
    console.error(error);
  } else {
    console.log('Scheduled omnimessage cancelled');
  }
});

File: code-samples/messaging/global-business/cancel-message.rb
require 'messente_api'

MessenteApi.configure do |config|
  config.username = 'YOUR_MESSENTE_API_USERNAME'
  config.password = 'YOUR_MESSENTE_API_PASSWORD'
end

api_instance = MessenteApi::OmnimessageApi.new

begin
  api_instance.cancel_scheduled_message('<omnimessage_id>')
  puts 'Scheduled omnimessage cancelled'
rescue MessenteApi::ApiError => e
  puts "Exception when cancelling an omnimessage: #{e}"
  puts e.response_body
end
